svc/api/admin/src/route/cluster/datacenter.rs

```diff
@@ -0,0 +1,106 @@
+use api_helper::ctx::Ctx;
+
+use proto::backend;
+use rivet_api::models;
+
+use rivet_operation::prelude::{proto::backend::pkg::cluster, *};
+
+use serde_json::{json, Value};
+
+use crate::auth::Auth;
+
+pub async fn create(
+	ctx: Ctx<Auth>,
+	cluster_id: Uuid,
+	body: models::AdminClusterDatacenterCreateRequest,
+) -> GlobalResult<models::AdminClusterDatacenterCreateResponse> {
+	let datacenter_id = Uuid::new_v4();
+
+	// Set the default drain timeout to 4 hours
+	let drain_timeout = 1000 * 60 * 60 * 4;
+
+	msg!([ctx] cluster::msg::datacenter_create(datacenter_id) -> cluster::msg::datacenter_scale {
+		datacenter_id: Some(datacenter_id.into()),
+		cluster_id: Some(cluster_id.into()),
+		name_id: body.name_id.clone(),
+		display_name: body.display_name.clone(),
+
+		provider: body.provider as i32,
+		provider_datacenter_id: body.provider_datacenter_id.clone(),
+		provider_api_token: None,
+
+
+
+		// When creating a datacenter, an empty pool of each type is added. This
+		// is to make sure that the datacenter starts in a valid state.
+		pools: vec![
+			backend::cluster::Pool {
+				pool_type: backend::cluster::PoolType::Job as i32,
+				hardware: vec![backend::cluster::Hardware {
+					provider_hardware: "g6-nanode-1".to_string(),
+				}],
+				desired_count: 0,
+				max_count: 0,
+				drain_timeout,
```

Comment:
doesn't use body.drain_timeout

Reply:
Changing this to remove `drain_timeout` from the body + cli, so that changing it
is moved to editing each datacenter instead of setting it here.